### PR TITLE
Changes (R&D made) Saber SMG to Advanced SMG

### PR DESCRIPTION
**Pull request by SpadesNeil**
This is more than just a name change, hence it being separate to that teeny tweak to the ammo count in the other PR.

If this PR gets in though, https://github.com/PolarisSS13/Polaris/pull/5878 is redundant and should be closed.

Okay, so, the Saber SMG right now is inconsistent and filled with redundant code and claims that it does things which it doesn't actually do.

For one, it doesn't actually have any AP ammo when spawned in R&D, and it never has. Without that, it's pointless to produce, and offers nothing special and unique. I altered it slightly to make it special and unique by giving it slightly better burst fire accuracy, and also, it can now accept standard 9mm magazines.

Two, it was called the "Saber" SMG, and you already have a Saber SMG in game, and this was really confusing. Security's top-loaded SMG is not the same as this one. This one is a teeny bit more accurate.

Oh, and I deleted that 9mm ammo box from the game, because that thing never made sense. Now, the silver you use to produce magazines for the gun make _some_ sense, because the silver needs to be used in proprietary magazines or something.

Also, the diamond needed to make the SMG is for the fancy scope.

Made more sense to me, anyway.

Last of all, the gun **no longer spawns with ammo** out of the protolathe. It now spawns unloaded by default.

I forgot to mention that the loaded and unloaded sprites will function for this now as well.

## tl;dr makes the Advanced SMG _actually worth making_

This was originally part of https://github.com/PolarisSS13/Polaris/pull/5837

### CODE REVIEW

**Comment by SpadesNeil** [comment]
Ah shit I just realized I didn't include the icons for this. They're in the other PR.

I'll get them later. Kind of burnt out right now.

**Comment by SpadesNeil** [comment]
So, I kind of fucked myself with the icons, because I have several things that involve icon changes, and I stupidly tried to include them all in one PR to avoid conflicts.

It looks like we're just going to have conflicts, unless Github no longer hates .dmi files, _or_ I can hold off on the icon changes until after I learn the status of several other PRs, and whether or not they will be accepted.

Among those PRs is
https://github.com/PolarisSS13/Polaris/pull/5875
https://github.com/PolarisSS13/Polaris/pull/5881 (this one)
and eventually one for the caseless rifle because some staff have expressed interest in that.

Since this isn't a **super** important change, I would suggest we hold off on this until these other PRs are resolved, so we can do all the icon stuff in one go rather than deal with many merge conflicts across several PRs.

Or not. It's up to you. Whatever sounds easier.

**Comment by Atermonera** [comment]
This is the SMG produced by R&D

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 1]
```diff
@@ -13,29 +13,32 @@
...
 
-/obj/item/weapon/gun/projectile/automatic/saber //Fixed it
-	name = "prototype SMG"
-	desc = "A protoype lightweight, fast firing gun. Uses 9mm rounds."
+/obj/item/weapon/gun/projectile/automatic/advanced_smg
+	name = "advanced SMG"
+	desc = "An advanced submachine gun with a reflective laser optic that makes burst fire less inaccurate than other SMGs. Uses 9mm rounds."
 	icon = 'icons/obj/gun.dmi'
-	icon_state = "saber"//Still ugly
+	icon_state = "advanced_smg-empty"
```
[the remark this answers is not in the document]
Comment (SpadesNeil) [reply code/modules/projectiles/guns/projectile/automatic.dm]: See my comment above. I need to correct the sprites, but since I have like three things that muck with the gun.dmi file in the pipeline, I'm trying to work out first what stuff is going to get accepted and what won't.

Otherwise, each of these PRs will conflict with one-another. It'll save some headache and lessens risks of breaking icons if I can do all the icon updates at once after knowing what stays and what goes.

[the remark this answers is not in the document]
Comment (SpadesNeil) [reply code/modules/projectiles/guns/projectile/automatic.dm]: Actually, I'm just going to nix the sprite changes to the tommygun PR as they aren't necessary.

So it's just going to be this thing and the G11 coming in a PR I plan to redo today.

**Comment by SpadesNeil** [comment]
>This is the SMG produced by R&D

Correct. The R&D SMG is the original "Saber" machine gun, but Polaris added the WT-220 "Saber" and there's always been two "Saber" SMGs ever since.

This makes it so there's only one Saber SMG, by making the R&D one a slightly more advanced prototype.

File: code/modules/research/designs/weapons.dm [hunk 2]
```diff
@@ -50,14 +50,14 @@
 	desc = "An compact reliable SMG firing armor piercing ammo."
```
Comment (SpadesNeil) [inline code/modules/research/designs/weapons.dm]: Whoops! I forgot to change this!!

File: code/modules/projectiles/guns/projectile/automatic.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (SpadesNeil) [reply code/modules/projectiles/guns/projectile/automatic.dm]: https://github.com/PolarisSS13/Polaris/pull/5886 makes icon changes that would cause conflicts in this PR, or vise-versa.

[the remark this answers is not in the document]
Comment (SpadesNeil) [reply code/modules/projectiles/guns/projectile/automatic.dm]: Screw it I'll just change them now and fix conflicts later.

**Comment by SpadesNeil** [comment]
I have manually merged the auto shotgun sprite from https://github.com/PolarisSS13/Polaris/pull/5906
